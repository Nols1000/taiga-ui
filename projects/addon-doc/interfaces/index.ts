export * from './api-template';
export * from './code-editor';
export * from './demo-params';
export * from './documentation-property';
export * from './page';
export * from './source-code-path-options';
