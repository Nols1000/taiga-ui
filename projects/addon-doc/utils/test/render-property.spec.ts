import {TuiDocumentationPropertyType} from '@taiga-ui/addon-doc';

describe(`tuiRenderProperty`, () => {
    const types: TuiDocumentationPropertyType[] = [
        `input`,
        `input-output`,
        `output`,
        null,
    ];

    const values = [`value`, undefined];

    beforeEach(() => {
        jest.spyOn(console, `error`);
    });

    describe.each(types)(`with type %s`, type => {
        it.each(values)(`with value %s, should render correct code`, async value => {
            jest.resetModules();
            const {tuiRenderProperty} = await import(`../render-property`);

            expect(
                tuiRenderProperty(`name`, {
                    type,
                    value,
                }),
            ).toMatchSnapshot();
        });
    });

    it(`should rethrow uncaught exception`, async () => {
        jest.mock<typeof import('../attr-name')>(`../attr-name`, () => ({
            tuiGetAttrName(): string {
                throw new Error(`Uncaught error`);
            },
        }));
        jest.resetModules();
        const {tuiRenderProperty} = await import(`../render-property`);

        expect(() =>
            tuiRenderProperty(`name`, {
                type: `input`,
                value: `test`,
            }),
        ).toThrow(`Uncaught error`);
    });
});
