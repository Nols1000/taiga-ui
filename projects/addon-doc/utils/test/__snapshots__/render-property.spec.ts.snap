// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tuiRenderProperty with type input with value undefined, should render correct code 1`] = `undefined`;

exports[`tuiRenderProperty with type input with value value, should render correct code 1`] = `"[name]="value""`;

exports[`tuiRenderProperty with type input-output with value undefined, should render correct code 1`] = `undefined`;

exports[`tuiRenderProperty with type input-output with value value, should render correct code 1`] = `"[(name)]="value""`;

exports[`tuiRenderProperty with type null with value undefined, should render correct code 1`] = `"name"`;

exports[`tuiRenderProperty with type null with value value, should render correct code 1`] = `"name="value""`;

exports[`tuiRenderProperty with type output with value undefined, should render correct code 1`] = `undefined`;

exports[`tuiRenderProperty with type output with value value, should render correct code 1`] = `"(name)="value""`;
